bigtable/spark/build.sbt

```diff
@@ -0,0 +1,42 @@
+name := "bigtable-spark-samples"
+
+version := "0.1"
+
+scalaVersion := "2.11.12"
+
+libraryDependencies ++= Seq(
+  "org.apache.spark" %% "spark-sql" % "2.4.5" % Provided,
+  "org.apache.hbase.connectors.spark" % "hbase-spark" % "1.0.0" % Provided,
+  // Exception: Incompatible Jackson 2.9.2
+  "com.fasterxml.jackson.module" %% "jackson-module-scala" % "2.9.10",
+  // NoClassDefFoundError: org/apache/spark/streaming/dstream/DStream
+  // when saving a DataFrame (!)
+  "org.apache.spark" %% "spark-streaming" % "2.4.5" % Provided,
+  "com.google.cloud.bigtable" % "bigtable-hbase-2.x-hadoop" % "1.15.0",
+  // NoClassDefFoundError: org/apache/hadoop/hbase/fs/HFileSystem
+  // Why?!?! The example does NOT use them directly!
+  "org.apache.hbase" % "hbase-server" % "2.2.3",
+  "org.apache.hbase" % "hbase-client" % "2.2.3"
+)
+
+excludeDependencies ++= Seq(
```

Comment:
Left a comment. Done.

Reply:
Needs more explanation: which deps already provide this?
Without this information how will we know that this can be removed when updating versions

Reply:
I didn't care what deps provide each and every excluded dependency. It was required to get `assembly` working (it failed due to multiple dependencies providing the same packages and classes). It was kind of trial and error way of solving it. Once it worked I crossed that task out.

Do you want me to dig deeper and describe the dependency "chains"? I'd argue if that helped much as "when updating versions" other deps may have to be excluded. WDYT?

Reply:
At the very least I would like a comment explaining what kind of issues these are causing? is it at build time? or run time? So that when we need to upgrade dependencies I know if these exclusions can be removed.

Reply:
There are the [comments](https://github.com/jaceklaskowski/java-docs-samples/blob/37d5e6402abcf2b0191edb1190957f629559a95c/bigtable/spark/build.sbt#L33-L34) already:

> // Excluding duplicates for the uber-jar
> // There are other deps to provide necessary packages

Do you want me to elaborate further?

Reply:
Yes, I saw those comments, but I need more specific information so that we know if we can remove these exclusion in future. I need to know what issue or error this is causing and when that error is occuring and why is sbt is not doing the right thing for me out of the box

Right now I can remove the entire exclusion list and everything looks like its working fine. How will I know in the future that these manual interventions are not necessary?